packages/block-editor/src/components/block-list/use-block-props/use-block-screen-reader-description.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * WordPress dependencies
+ */
+import { sprintf, __ } from '@wordpress/i18n';
+import { getBlockType } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../../store';
+
+export function useBlockScreenReaderDescription( clientId ) {
+	const { hasChildBlocks, blockTitle } = useSelect(
+		( select ) => {
+			const { getBlockRootClientId, getBlockName, getBlock } = select(
+				blockEditorStore
+			);
+			const clientIdToUse = getBlockRootClientId( clientId );
+			const blockName = getBlockName(
+				clientIdToUse ? clientIdToUse : clientId
+			);
+			const blockType = getBlockType( blockName );
+			return {
+				hasChildBlocks:
+					getBlock( clientIdToUse ? clientIdToUse : clientId )
+						?.innerBlocks?.length > 0,
+				blockTitle: blockType?.title,
+			};
+		},
+		[ clientId ]
+	);
+	let description;
+	if ( hasChildBlocks ) {
+		description = sprintf(
+			// Translators: 1: The block title to lowercase for good sentence structure.
+			__( 'Child block of %1$s.' ),
```

Comment:
@youknowriad @talldan Maybe I should use getBlocks() instead so I can compare the index? The issue is, I cannot tell if I am in a parent block or not. E.g. columns block with equal split column blocks. If I select the columns block, I hear "Child of columns." Can I, in code, detect the difference between parent block and where child blocks actually start?

The other thing to work out is detecting if a block supports child blocks. For example, the paragraph block. In another update, I want to exclude the paragraph block or any block that doesn't have support for child blocks. That way I can add instructions for say the group/columns blocks without having to hear those announced for the paragraph block. Is there some code I can use to see if a block supports child blocks?

Reply:
I think I have this fixed up now. Logic may need help.